include/clap/helpers/plugin.hh

```diff
@@ -315,6 +331,11 @@ namespace clap { namespace helpers {
       void ensureAudioThread(const char *method) const noexcept;
       void ensureParamThread(const char *method) const noexcept;
 
+      ////////////////////
+      // General Checks //
+      ////////////////////
+      void ensureIsInactive(const char *methodName) const noexcept;
```

Comment:
You should also add `ensureIsActive()` for completeness.

Reply:
Also if you add those two method, then you should update the other place to use them for consistency.

Reply:
> You should also add ensureIsActive() for completeness.

Done via fe1d73c31b6677744dfd166b8ec9e84d2e6863ba.  

> Also if you add those two methods, then you should update the other place to use them for consistency.  

I'm unsure if I understood this right. If you want the new functions `ensureIsInactive()` and `ensureIsActive()` to be used in the entire `plugin.hxx` where appropriate, I think that's generally a good idea. But I also think that introducing these new functions outside of `clapConfigurableAudioPortsCanApplyConfiguration()` and `clapConfigurableAudioPortsApplyActivation()` would be out of scope for this MR which is about adding a base implementation for the configurable audio ports extension.

Reply:
If you don't do it now, then you'll leave the code in an inconsistent state.

Reply:
I think we could replace the following uses with the new functions:
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L288-L289 (error message would be a bit different)
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L335
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L359-L361 (error message would be a bit different)
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L377-L379 (error message would change just slightly)
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L400-L402 (error message would change just slightly)
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L423-L424 (error message would change just slightly)
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L440-L441 (error message would change just slightly)
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L757
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L1297-L1299

With these uses, I don't think replacing it with the new functions would make sense:
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L230-L235 (not appropriate: if line 233 failed in the plugin, then the failing assert in line 235 will be the plugin's fault and not the host's fault)
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L341 (this error would be the plugin's fault while the new function calls misbehaving host)
- https://github.com/Schroedingers-Cat/clap-helpers/blob/next/include/clap/helpers/plugin.hxx#L549-L552 (error message would be less explanatory with the replacement)

Let me know if that's what you requested.

Reply:
Actually it won't work so well, because of how the error message is made and how we handle the error.
Maybe better skip this `ensureIsActive()` change.

Reply:
Alright!